### ISSUE TITLE
Function space was correct, but eslint show 'Unexpected space between function name and paren'

### ISSUE THREAD

**Issue by susiwen8** [eslint/eslint#13598]
**Tell us about your environment**
macOS High Sierra version 10.13.6

Environment Info:

Node version: v12.14.0
npm version: v6.14.5
Local ESLint version: v7.7.0 (Currently used)
Global ESLint version: Not found

**What parser (default, Babel-ESLint, etc.) are you using?**
@typescript-eslint/parser

**Please show your full configuration:**

<details>
<summary>

parser: "@typescript-eslint/parser"
parserOptions:
    project: "tsconfig.json"
plugins: ["@typescript-eslint"]
rules:
    no-spaced-func: 2

```js

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```ts
function a<T, R>(): number {
    return 1;
}

const b = a<{
    startValue: number | (string | number)[],
    nextValue: number | (string | number)[]
}, string>();

const c = a<{startValue: number | (string | number)[], nextValue: number | (string | number)[]}, string>();

```

```bash
./node_modules/.bin/eslint test.ts
```

**What did you expect to happen?**
No error about `Unexpected space between function name and paren. eslint(no-spaced-func)`

**What actually happened? Please include the actual, raw output from ESLint.**
`error  Unexpected space between function name and paren  no-spaced-func`

![Screen Shot 2020-08-21 at 10 03 37](https://user-images.githubusercontent.com/20318608/90844063-a10f7b00-e395-11ea-833f-1889b897e0c1.png)

**Are you willing to submit a pull request to fix this bug?**
If someone guide me, I'm willing to submit a pull request

PS:
```ts
const b = a<{
    startValue: number |(string | number)[],
    nextValue: number | (string | number)[]
}, string>();
```
This code won't trigger `no-space-func`

**Comment by yeonjuan** [comment]
HI :) @susiwen8 

`no-spaced-func` rule was **deprecated** in ESLint v3.x. And it replaced with [func-call-spacing](https://eslint.org/docs/rules/func-call-spacing) but this rule also does not support typescript syntax.

So I suggest using the [@typescript-eslint/func-call-spacing](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/func-call-spacing.md).

**Comment by susiwen8** [comment]
@yeonjuan Thanks for the tip
